### ISSUE TITLE
`no-warning-comments` should include the comment itself in the report (UX only)

### ISSUE THREAD

**Issue by fregante** [eslint/eslint#12327]
I think the point of this rule is to keep track of TODO comments, however this is hard to follow; I'd have to open each file separately to see if I can handle a task.

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment.                                                                   no-warning-comments
  ⚠    2:1   Unexpected todo comment.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   Unexpected todo comment.                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   Unexpected todo comment.                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   Unexpected todo comment.                                                                   no-warning-comments
```

**What rule do you want to change?**

`no-warning-comments`

**Does this change cause the rule to produce more or fewer warnings?**

Unchanged. It's only about UX

**How will the change be implemented? (New option, new default behavior, etc.)?**

New default

**What does the rule currently do for this code?**

Shown above

**What will the rule do after it's changed?**

```ts
  source/globals.d.ts:1:1
  ⚠    1:1   FIXME: Drop ignore after #9.                                                                  no-warning-comments
  ⚠    2:1   FIXME: Find this info on the page.                                                                   no-warning-comments

  source/libs/post-form.ts:9:3
  ⚠    9:3   TODO: Improve JSX types for event listeners so we can use... [clipped]                                                                   no-warning-comments

  source/features/hidden-review-comments-indicator.tsx:37:2
  ⚠   37:2   TODO: Drop some definitions when their related bugs are resolved                                                                   no-warning-comments

  source/features/revert-file.tsx:89:4
  ⚠   89:4   TODO: Drop `as` after https://github.com/microsoft/TSJS-lib-generator/issues/741                                                                   no-warning-comments
```

**Comment by g-plane** [comment]
I'll champion this.

**Comment by fregante** [comment]
Great!

**Comment by g-plane** [comment]
We need at least 3 upvotes from @eslint/eslint-team to accept this enhancement.

**Comment by kaicataldo** [comment]
In general, I'm in support of this, though I do think we want to have some kind of prepended message so it's clear what is being warned about.

**Comment by platinumazure** [comment]
Should we truncate the comment in case it's really long?

Should we show only the first line in a multiple-line block comment?

**Comment by anikethsaha** [comment]
I think we can set a number of char to show in the message report followed by `...` . 

I think 30-40 chars limit can be good.

**Comment by nzakas** [comment]
@g-plane as the champion, it's up to you to get the support of others on the ESLint team. It looks like we are still missing two more :+1:s.

**Comment by g-plane** [comment]
@eslint/eslint-team Would we like to support this?

**Comment by anikethsaha** [comment]
+1 for this. 

I can think of something like this 

```
  source/globals.d.ts:1:1
  ⚠    1:1   Unexpected todo comment: TODO: something short                                                     no-warning-comments
  ⚠    2:1   Unexpected todo comment: TODO: something really long...                                            no-warning-comments

```

**Comment by mdjermanovic** [comment]
There are 3 :+1: from team members now, so marking as accepted.

**Comment by anikethsaha** [comment]
@g-plane are you working in this?

**Comment by g-plane** [comment]
Currently, no.

**Comment by anikethsaha** [comment]
I can work on this if it's ok?

**Comment by g-plane** [comment]
Sure. Feel free to do it.
